jordyob03/TripTailor
# TT 62 LG searching

# Summary

Added bare bones implementation for a searching service. The structure closely models the auth service

- Implemented search functionality for itineraries based on location (city and country).

---

## Changes

- [x] Added /search: This endpoint accepts country and city as query parameters and returns the list of matching itineraries using SQL.
- [x] Connected the service to PostgreSQL for storing and retrieving itineraries.
- [x] Miguel has added the Dockerfile
- [x] Abdul has added some basic front end code (**Search.js, SearchAPI.js**)  

---

## Implementation Details

Axios Integration (Frontend):
	•	The frontend uses Axios to send a GET request with query parameters **(Search.js)**:
                `const response = await searchAPI.get('/search', { params: { country, city } });`

---

## How to Test

To use the service locally, you will need to have sample data in your **_itineraries_** and _**users**_ tables. The queries i used are located in the **test** folder. **Do not try to run these queries from your IDE**.  Instead, copy them and run them through pgadmin.

Then, run the service:
   ```bash
   docker-compose up -d search-service
```
Enter Search (make sure this search is valid given the data you have in your tables) @ http://localhost:3000/search
<img width="264" alt="image" src="https://github.com/user-attachments/assets/8a8153cd-3cca-4497-a4c8-573637569084">


Then view the response:
<img width="574" alt="image" src="https://github.com/user-attachments/assets/68836e59-02ca-4b88-a99a-aa05b0d2ca26">
